### ISSUE TITLE
bisync requires --resync after login failure

### ISSUE THREAD

**Issue by LivingUserError** [rclone/rclone#7603]
#### What is the problem you are having with rclone?
In case a login fails, including due to an expired token or network timeout, it seems like `--resync` is required to continue syncing, since the path listings are deleted/renamed in the process. This is true even when using `--resilient`. I observed this behaviour in both cases using Proton Drive. I would argue that neither a timeout nor an expired token should cause this behaviour, because it is not a problem with the sync itself, but the limited reliability of network syncs. At least in resilient-mode, I would expect rclone to just abort with an error and allow for retries.

#### What is your rclone version (output from `rclone version`)
rclone v1.65.1
- os/version: arch rolling (64 bit)
- os/kernel: 6.7.1-arch1-1 (x86_64)
- os/type: linux
- os/arch: amd64
- go/version: go1.21.6
- go/linking: dynamic
- go/tags: none

#### Which OS you are using and how many bits (e.g. Windows 7, 64 bit)
Arch Linux, 64 bit

#### Which cloud storage system are you using? (e.g. Google Drive)
Proton Drive

#### The command you were trying to run (e.g. `rclone copy /tmp remote:tmp`)
`rclone bisync proton:sync /home/user/proton "--protondrive-replace-existing-draft=true" "--resilient" "--create-empty-src-dirs" "--refresh-times" "--metadata" "--transfers" "8" "--human-readable" &`

#### A log from the command with the `-vv` flag (e.g. output from `rclone -vv copy /tmp remote:tmp`)
```shell
2024/01/27 20:24:01.218750 WARN RESTY 401 GET https://mail.proton.me/api/core/v4/users: Invalid access token (Code=401, Status=401), Attempt 1
2024/01/27 20:24:01.218809 ERROR RESTY 401 GET https://mail.proton.me/api/core/v4/users: Invalid access token (Code=401, Status=401)
2024/01/27 20:24:01.602754 WARN RESTY 401 GET https://mail.proton.me/api/core/v4/users: Invalid access token (Code=401, Status=401), Attempt 1
2024/01/27 20:24:01.602769 ERROR RESTY 401 GET https://mail.proton.me/api/core/v4/users: Invalid access token (Code=401, Status=401)
2024/01/27 20:24:04.311530 WARN RESTY 422 POST https://mail.proton.me/api/auth/v4/2fa: Incorrect login credentials. Please try again. (Code=8002, Status=422), Attempt 1
2024/01/27 20:24:04.311546 ERROR RESTY 422 POST https://mail.proton.me/api/auth/v4/2fa: Incorrect login credentials. Please try again. (Code=8002, Status=422)
Failed to create file system for "proton:sync": couldn't initialize a new proton drive instance: 422 POST https://mail.proton.me/api/auth/v4/2fa: Incorrect login credentials. Please try again. (Code=8002, Status=422)
```

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by nielash** [comment]
Thanks for the report! I agree, and I think this should be resolved by the new [`--recover`](https://tip.rclone.org/bisync/#recover) feature, which is now available in the [latest beta](https://beta.rclone.org/?sort=time&order=desc). Would you want to give it a try and let me know if it solves this for you?

**Comment by LivingUserError** [comment]
Hi, thanks for the quick response. I don't really want to replace my current install with a beta package outside of the Arch repos (or AUR for that matter) for now, but I'll check whether everything works when 1.66 drops.

I have noticed there is `rclone-beta-bin` in the AUR, but it's v1.58. I will promise nothing, but maybe I'll take care if this at some point :D

**Comment by nielash** [comment]
Ok, sounds good! In the meantime I'm going to go ahead and optimistically close this, as I believe this is resolved by `v1.66`. But please feel free to reopen it if you find that there is still an issue, once you have a chance to try it!

> I have noticed there is `rclone-beta-bin` in the AUR, but it's v1.58.

Wow, nearly 2 years out of date! Definitely don't use that one 😂
